### ISSUE TITLE
Attribute Errors

### ISSUE THREAD

**Issue by brusehc** [felipeochoa/minecart#13]
Entry-level python programmer attempting to use minecart as shown in examples..
Installed minecart, pdfminer.six
Getting all sorts of attribute errors, slowly working through the documentation to remove errors

ex:
doc = minecart.Document(fp)

- result -
AttributeError: module 'pdfminer.pdfparser' has no attribute 'PDFDocument'

then changed "miner.py", line 330 to:
self.doc = pdfminer.pdfdocument.PDFDocument(self.parser, caching=True)

- this fixed the problem, and these two lines run properly...

doc = minecart.Document(fp)
doc2 = doc.iter_pages()

- but now.....

page = doc.get_page(0)

- result -
  File "pdf_reader.py", line 28, in <module>
    page = doc.get_page(0)
  File ".../minecart/miner.py", line 350, in get_page
    for i, page in enumerate(self.doc.get_pages()):
AttributeError: 'PDFDocument' object has no attribute 'get_pages'

Is there a problem in the way the environment has been configured, or the versions of the packages that I'm using? I would greatly appreciate any help as I really want to use minecart !

**Comment by felipeochoa** [comment]
You need to use pdfminer3k, not pdfminer.six.

**Comment by brusehc** [comment]
Thanks for the timely response!
I have installed pdfminer3k as well. Is there a way I can make sure it is being used properly?

**Comment by felipeochoa** [comment]
Make sure to uninstall pdfminer.six, something like `pip uninstall pdfminer.six` minecart should work out of the box in that setup. You may have to reinstall it if you've modified any files

**Comment by brusehc** [comment]
OK, great! minecart is up an running. One more question: when opening a pdf file that requires decryption, how should I include the password?

**Comment by felipeochoa** [comment]
Not sure. You're probably better off removing the password protection using something like `pdftk` first. I'm not sure if pdfminer supports password protected pdfs

**Comment by cyberbruno** [comment]
Hello, I have the same error :AttributeError: module 'pdfminer' has no attribute 'pdfdocument'
Try to uninstall all PDF modules and re-install them making sure no pdfminer.six was on the computer...
NO luck. Anything I should doublecheck?

**Comment by felipeochoa** [comment]
In a python shell import pdfminer and see what file is coming from. Something like:

```
Import pdfminer
print(pdfminer.__file__)
```

**Comment by cyberbruno** [comment]
I got this:
Exception has occurred: AttributeError
module 'pdfminer' has no attribute '_file_'

**Comment by cyberbruno** [comment]
and I have pdfminer3k==1.3.1; minecart==0.3.0, and PyPDF2==1.26.0.
Note that I ran the exact command you suggest.

**Comment by felipeochoa** [comment]
Sorry I'm on my phone and can't test, but can you ensure the underscores are in there? If still not working you'll have to search for how to confirm. Unless pdfminer 3k has changed recently, you'll only get that error from using the wrong pdfminer library

**Comment by bjornasm** [comment]
Still having problems with this, even after removing pdfminer3k, pdfminer.six and minecart. Can we get an update on this?

**Comment by pptyasar** [comment]
Did you got any solution @bjornasm  @cyberbruno

**Comment by kausarahmad** [comment]
I had the same issue. I uninstalled pdfminer3k, pdfminer.six, pdfminer, pypdf2, and minecart. Then, I reinstalled minecart. Still had no luck.

I fixed it by uninstalling the poppler package (conda uninstall poppler) and then reinstalling minecart.

**Comment by SuperPauly** [comment]
I too face "{AttributeError}module 'pdfminer.pdfparser' has no attribute 'PDFDocument'". I have reinstalled PDFminer3k 3 times now. Have reinstalled minecart 3 times aswell. I have uninstalled PDFMiner.six and I still get this. 

I can't get access to miner.py to chnge the file becuase of permissions set. I'm on Win 10. Whats even more anoying I can get to the directory where miner.py exists becuase Win 10 is telling me the directory don't exist.
